pkg/envoy/rds/response.go

```diff
@@ -56,5 +58,40 @@ func NewResponse(cataloger catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_dis
 		rdsResources = append(rdsResources, ingressRouteConfig)
 	}
 
+	if dr != nil {
```

Comment:
When would this be nil?

Reply:
Actually.... nvm. Unit tests are assuming nil, but this makes the tests wrong. I'll have to fix.

Reply:
Working on this.... will take a while to fix unit tests